### PR TITLE
feat(deno): update spec to 1.16

### PR DESCRIPTION
**Pull request by SeparateRecords**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Spec update

**What is the current behavior? (You can also link to an open issue here)**
Not up to date.

**What is the new behavior (if this is a feature change)?**
Updates Deno spec to 1.15 & 1.16. This modernises the spec a bit (sets `requiresEquals` where appropriate) and, because requiresEquals optional arguments are supported now, adds an environment variable generator for `--allow-env`.

**Additional info:**
Deno 1.16 technically isn't out yet, but will be in less than 24 hours. It doesn't actually introduce anything in the CLI, all the new stuff was in 1.15.
(Still saying it's updated to 1.16 so future-me doesn't get confused when updating to 1.17)

Also, I've been gone for a while for a bunch of _really_ stupid reasons that I've resolved now, all to do with my flaky setup. Hopefully I'll be back to being active in this community :)

### CODE REVIEW

**Comment by SeparateRecords** [comment]
Hah! 1.16 was released _while I was writing the PR description_.

image

**Comment by SeparateRecords** [comment]
Reverting to draft - didn't realise that Fig spawns a new shell so `env` won't show the user's variables. Will fix tomorrow

File: src/deno.ts [hunk 1]
```diff
@@ -4,9 +4,6 @@
 // All objects marked with '// requiresEquals: true' are Clap args with '.require_equals(true)'
 // TODO: When fig supports this option (or something like it), uncomment the arguments.
```
Comment (QuiiBz) [inline src/deno.ts:5]: Can we remove this TODO comment?

File: src/deno.ts [hunk 2]
```diff
@@ -811,7 +829,6 @@ const denoDoc: Fig.Subcommand = {
               throw new Error(`Output data was JSON, but was not an array`);
             }
           } catch (err) {
-            console.error("Returning early due to error:", err);
```
Comment (QuiiBz) [inline src/deno.ts:814]: That can still be useful if anything goes wrong, to make debugging easier

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for this contribution!

File: src/deno.ts (hunk 1 above)
Comment (SeparateRecords) [reply src/deno.ts:5]: Thanks for noticing that. I removed that suggestion from the find&replace results and totally forgot to remove it 😆

**Review by QuiiBz — APPROVED** [review]
LGTM, thanks for keeping this spec up to date!

**Pull request merged**